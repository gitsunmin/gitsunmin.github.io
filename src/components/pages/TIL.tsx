type Props = React.PropsWithChildren;

export const TILPage = ({ children }: Props) => (
  <div className="px-6 pt-4 md:px-0">{children}</div>
);
