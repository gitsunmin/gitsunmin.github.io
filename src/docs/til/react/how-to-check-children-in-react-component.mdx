## How to check children in React Component

```typescript
import React from 'react';

const ComponentA = ({ children }) => {
  let hello;
  React.Children.forEach(children, (child) => {
    console.log(child);
    /**
     * {
     * 		$$typeof: 	Symbol(react.element)
     * 		key		: 	component에 설정한 key
     * 		props	:	component에 설정한 props
     * 		ref		: 	component에 설정한 ref
     * 		type	: 	component의 정보를 담고있는 객체
     * }
     **/
  });

  return hello;
};
```