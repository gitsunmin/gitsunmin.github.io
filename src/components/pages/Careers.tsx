import { Career } from '@/data/careers';
import { Suspense } from 'react';

const Content = () => {
  return (
    <ul className="flex flex-col gap-8 pt-8">
      {Career.map((career) => {
        return (
          <li
            key={career.name}
            className="p-6 rounded-xl border shadow-sm bg-background text-foreground flex flex-col gap-4"
          >
            <div className="flex items-start justify-start gap-x-4">
              <img
                src={career.logo}
                alt={`${career.name} 로고`}
                className="rounded-full size-12 border"
              />
              <div>
                <h2 className="text-xl font-bold">{career.name}</h2>
                <p className="text-sm text-gray-500 dark:text-gray-200 mt-1">
                  {career.positoin} | {career.range}
                </p>
              </div>
            </div>

            {/* 주의사항 */}
            {career.caution && (
              <p className="text-sm text-red-500">* {career.caution}</p>
            )}

            {/* 회사 소개 */}
            <section className="text-gray-700 dark:text-gray-300 leading-relaxed text-sm whitespace-pre-line">
              {career.introduce}
            </section>

            {/* 프로젝트 섹션 */}
            <section>
              <h6 className="mt-2 font-semibold text-sm text-gray-700 dark:text-gray-300">
                Experiencies
              </h6>
              <ul className="list-disc ml-5 text-sm mt-1">
                {career.experiencies.map(({ id, name }) => (
                  <li key={id}>{name}</li>
                ))}
              </ul>
            </section>

            {/* 링크 섹션 */}
            <section>
              <h6 className="mt-2 font-semibold text-sm text-gray-700 dark:text-gray-300">
                Links
              </h6>
              <ul className="list-disc ml-5 text-sm mt-1 space-y-1">
                {career.links.map(({ label, url }) => (
                  <li key={url}>
                    <a
                      href={url}
                      target="_blank"
                      rel="noopener noreferrer"
                      className="text-blue-600 hover:underline inline-flex items-center gap-1"
                    >
                      {label}
                      <span>↗</span>
                    </a>
                  </li>
                ))}
              </ul>
            </section>
          </li>
        );
      })}
    </ul>
  );
};

export const CareersPage = () => {
  return (
    <Suspense>
      <Content />
    </Suspense>
  );
};
