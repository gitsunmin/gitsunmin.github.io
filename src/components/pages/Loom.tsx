import { Suspense, useState } from 'react';
import { cn } from '@/lib/utils';

const Content = () => {
  const [bgColor, setBgColor] = useState('#FFFFFF');
  const [fgColor, setFgColor] = useState('#000000');

  const getBlackAndWhite = (bgColor: string): string => {
    if (!bgColor || typeof bgColor !== 'string') {
      return '#000000';
    }

    const hex = bgColor.startsWith('#') ? bgColor.slice(1) : bgColor;

    const r = Number.parseInt(hex.substring(0, 2), 16);
    const g = Number.parseInt(hex.substring(2, 4), 16);
    const b = Number.parseInt(hex.substring(4, 6), 16);

    if (Number.isNaN(r) || Number.isNaN(g) || Number.isNaN(b)) {
      return '#000000';
    }

    const yiq = (r * 299 + g * 587 + b * 114) / 1000;

    return yiq >= 128 ? '#000000' : '#FFFFFF';
  };

  const handleFixedColorChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const backgroundColor = e.currentTarget.value;
    const foregroundColor = getBlackAndWhite(backgroundColor);

    setBgColor(backgroundColor);
    setFgColor(foregroundColor);
  };

  return (
    <article className="px-4 mt-12 pt-4 w-full md:max-w-(--breakpoint-md) mx-auto">
      <section
        aria-label="color picker"
        className="relative rounded-lg"
        content="max-h-40"
      >
        <input
          type="color"
          aria-label="background color picker"
          value={bgColor}
          className="w-full h-40 rounded-lg border-2 p-0 cursor-pointer shadow-2xl [&::-webkit-color-swatch]:border-none [&::-webkit-color-swatch]:rounded [&::-webkit-color-swatch-wrapper]:p-px"
          onInput={handleFixedColorChange}
        />
        <div
          style={{ color: fgColor, borderColor: fgColor }}
          className={cn(
            'absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2',
            'border px-4 py-2 shadow-2xl rounded-lg text-center'
          )}
        >
          Gitsunmin is <br /> The Best Programer!
        </div>
      </section>
    </article>
  );
};

export const LoomPage = () => {
  return (
    <Suspense>
      <Content />
    </Suspense>
  );
};
