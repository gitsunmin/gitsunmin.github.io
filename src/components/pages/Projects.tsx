import { Canvas, useFrame, useThree } from '@react-three/fiber';
import { PROJECT_LIST } from '@/data/projects';
import { useRouter } from '@tanstack/react-router';
import { OrbitControls } from '@react-three/drei';
import { Suspense, useRef, useState } from 'react';
import { match } from 'ts-pattern';
import { BookShelf } from '@/components/canvas/BookShelf';
import * as THREE from 'three';

import BookTopTexture from '@/assets/book_top.webp';
import BookBottomTexture from '@/assets/book_bottom.webp';
import BookPagesTexture from '@/assets/book_pages.webp';

const books = PROJECT_LIST.filter((project) => project.parentId === null).map(
  (project) => ({
    id: project.id,
    textures: {
      front: project.book.cover.front,
      back: project.book.cover.back,
      side: project.book.cover.side,
      top: BookTopTexture,
      bottom: BookBottomTexture,
      pages: BookPagesTexture,
    },
  }),
);

function CameraController({
  selectedBookId,
}: {
  selectedBookId: string | null;
}) {
  const { camera } = useThree();
  const targetPosition = useRef(new THREE.Vector3(5, 5, 10));

  useFrame(() => {
    camera.position.lerp(targetPosition.current, 0.05); // 부드럽게 보간
    camera.lookAt(0, 0, 0);
  });

  // 선택된 책에 따라 목표 위치 변경
  if (selectedBookId) {
    targetPosition.current.set(0, 5, 18); // 정면
  } else {
    targetPosition.current.set(5, 5, 10); // 원래 위치
  }

  return null;
}

export const Content = () => {
  const router = useRouter();
  const [selectedBookId, setSelectedBookId] = useState<string | null>(null);

  const handleBookClick = (id: string) => {
    const wantToSeeDetails = match(selectedBookId)
      .with(id, () => confirm('해당 프로젝트를 자세히 보시겠습니까?'))
      .otherwise(() => false);

    if (wantToSeeDetails) {
      router.navigate({
        to: `/projects/${id}`,
      });
    }
    setSelectedBookId((prev) => (prev === id ? null : id));
  };

  return (
    <Canvas
      camera={{ position: [5, 5, 10], zoom: 2 }}
      style={{ height: '100dvh' }}
    >
      <CameraController selectedBookId={selectedBookId} />
      <ambientLight />
      <directionalLight position={[2, 5, 5]} />
      <BookShelf
        books={books}
        selectedBookId={selectedBookId}
        onBookClick={handleBookClick}
      />
      <OrbitControls enableZoom={false} />
    </Canvas>
  );
};

export const ProjectsPage = () => {
  return (
    <Suspense fallback={<div>Loading...</div>}>
      <Content />
    </Suspense>
  );
};
