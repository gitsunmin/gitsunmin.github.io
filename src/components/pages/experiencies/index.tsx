import { Canvas, useFrame, useThree } from '@react-three/fiber';
import { EXPERIENCE_LIST } from '@/data/experiencies';
import { useRouter } from '@tanstack/react-router';
import { OrbitControls } from '@react-three/drei';
import { Suspense, useRef, useState } from 'react';
import { match } from 'ts-pattern';
import { BookShelf } from '@/components/canvas/BookShelf';
import * as THREE from 'three';

import BookTopTexture from '@/assets/book_top.webp';
import BookBottomTexture from '@/assets/book_bottom.webp';
import BookPagesTexture from '@/assets/book_pages.webp';

const books = EXPERIENCE_LIST.map((experience) => ({
  id: experience.id,
  textures: {
    front: experience.book.cover.front,
    back: experience.book.cover.back,
    side: experience.book.cover.side,
    top: BookTopTexture,
    bottom: BookBottomTexture,
    pages: BookPagesTexture
  }
}));

function CameraController({
  selectedBookId
}: {
  selectedBookId: string | null;
}) {
  const { camera } = useThree();
  const targetPosition = useRef(new THREE.Vector3(5, 5, 10));

  useFrame(() => {
    camera.position.lerp(targetPosition.current, 0.05); // 부드럽게 보간
    camera.lookAt(0, 0, 0);
  });

  // 선택된 책에 따라 목표 위치 변경
  if (selectedBookId) {
    targetPosition.current.set(0, 5, 18); // 정면
  } else {
    targetPosition.current.set(5, 5, 10); // 원래 위치
  }

  return null;
}

export const Content = () => {
  const router = useRouter();
  const [selectedBookId, setSelectedBookId] = useState<string | null>(null);

  const handleBookClick = (id: string) => {
    const wantToSeeDetails = match(selectedBookId)
      .with(id, () => confirm('해당 프로젝트를 자세히 보시겠습니까?'))
      .otherwise(() => false);

    if (wantToSeeDetails) {
      router.navigate({
        to: `/experiencies/${id}`
      });
    }
    setSelectedBookId((prev) => (prev === id ? null : id));
  };

  return (
    <Canvas
      camera={{ position: [5, 5, 10], zoom: 2 }}
      className="h-[calc(100vh-64px)] w-full"
      style={{ height: window.innerHeight - 64 }}
    >
      <CameraController selectedBookId={selectedBookId} />
      <ambientLight />
      <directionalLight position={[2, 5, 5]} />
      <BookShelf
        books={books}
        selectedBookId={selectedBookId}
        onBookClick={handleBookClick}
      />
      <OrbitControls enableZoom={false} />
    </Canvas>
  );
};

export const ExperienciesPage = () => {
  return (
    <Suspense>
      <Content />
    </Suspense>
  );
};
