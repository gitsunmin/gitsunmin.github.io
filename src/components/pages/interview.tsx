import { Suspense } from 'react';
import { ChatCard } from '@/components/Chat';
import { INTERVIEW } from '@/data/interview';

import { cn } from '@/lib/utils';
import { Avatar } from '@/components/Avatar';

const Content = () => {
  return (
    <article className="flex flex-col text-gray-900 dark:text-gray-100">
      <p className="text-gray-500 dark:text-gray-400 border-l-2 border-gray-300 dark:border-gray-600 px-4 py-2 text-sm break-words bg-gray-50 dark:bg-gray-800">
        이 페이지는 제가 지금까지 면접이나 주변 사람들에게 들었던 질문들과 그에
        대한 답변을 기반으로 작성해보았습니다.
      </p>
      <section className="flex flex-wrap gap-y-[10px] pt-10">
        {INTERVIEW.map((chat, index, list) => {
          const { name } = chat;
          const isContinues = list[index - 1]?.name === name;

          return (
            <div
              // biome-ignore lint/suspicious/noArrayIndexKey: <explanation>
              key={`${name}-${index}`}
              className={cn('w-full flex gap-x-2.5 gap-y-1', {
                'flex-row': name === 'James',
                'flex-row-reverse': name === 'Sunmin',
              })}
            >
              {isContinues ? (
                <div className="w-10" />
              ) : (
                <div className="flex items-start">
                  <Avatar
                    src={
                      name === 'James'
                        ? 'https://images.unsplash.com/photo-1736033302187-64b6be9e4652?q=80&w=3474&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D'
                        : 'https://github.com/gitsunmin.png'
                    }
                    alt={`${name}_profile`}
                    size={40}
                    className="rounded-full border border-gray-200 dark:border-gray-700"
                  />
                </div>
              )}
              <div className={'max-w-[60%] flex flex-col gap-y-1'}>
                {isContinues ? null : (
                  <div
                    className={cn('text-sm text-gray-700 dark:text-gray-300', {
                      'text-left': name === 'James',
                      'text-right': name === 'Sunmin',
                    })}
                  >
                    {name}
                  </div>
                )}
                <ChatCard {...chat} />
              </div>
            </div>
          );
        })}
      </section>
    </article>
  );
};

export const InterviewPage = () => {
  return (
    <Suspense>
      <Content />
    </Suspense>
  );
};
