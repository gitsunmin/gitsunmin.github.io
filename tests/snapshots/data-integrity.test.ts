import { describe, it, expect } from 'vitest';
import { CareersPage } from '@/components/pages/Careers';
import { render } from '@testing-library/react';

describe('data-integrity', () => {
  it('Careers Page Snapshot Test', () => {
    expect(render(CareersPage())).toMatchSnapshot();
  });
});
