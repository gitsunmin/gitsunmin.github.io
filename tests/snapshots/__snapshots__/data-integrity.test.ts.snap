// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`data-integrity > Careers Page Snapshot Test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="flex flex-col gap-8 pt-8"
      >
        <li
          class="p-6 rounded-xl border shadow-sm bg-background text-foreground flex flex-col gap-4"
        >
          <div
            class="flex items-start justify-start gap-x-4"
          >
            <img
              alt="(주) 마켓보로 로고"
              class="rounded-full size-12 border"
              src="/src/assets/logos/marketboro_logo.webp"
            />
            <div>
              <h2
                class="text-xl font-bold"
              >
                (주) 마켓보로
              </h2>
              <p
                class="text-sm text-gray-500 dark:text-gray-200 mt-1"
              >
                프론트엔드 개발자
                 | 
                2020.10 ~ 
              </p>
            </div>
          </div>
          <section
            class="text-gray-700 dark:text-gray-300 leading-relaxed text-sm whitespace-pre-line"
          >
            (주) 마켓보로는 B2B 식자재 유통 시장에서 중간 과정을 최소화하여 비용 절감과 효율 증대를 실현하는 엔터프라이즈 푸드테크 플랫폼을 제공합니다. 
          또한 사업자 대상의 오픈 마켓인 식봄을 운영하고 있습니다. 
          저는 프론트엔드 개발자로서 React.js와 Next.js를 활용하여 웹과 모바일 애플리케이션의 사용자 인터페이스를 개발하고 있습니다. 
          특히, 기존 PHP로 개발된 식봄을 Next.js로 전환하는 프로젝트를 주도하며, 성능 최적화와 사용자 경험 향상을 달성하였습니다. 
          이러한 경험을 통해 다양한 디바이스에서의 접근성을 보장하고, 최신 기술 스택을 적극 활용하는 역량을 갖추게 되었습니다.
          </section>
          <section>
            <h6
              class="mt-2 font-semibold text-sm text-gray-700 dark:text-gray-300"
            >
              Projects
            </h6>
            <ul
              class="list-disc ml-5 text-sm mt-1"
            />
          </section>
          <section>
            <h6
              class="mt-2 font-semibold text-sm text-gray-700 dark:text-gray-300"
            >
              Links
            </h6>
            <ul
              class="list-disc ml-5 text-sm mt-1 space-y-1"
            >
              <li>
                <a
                  class="text-blue-600 hover:underline inline-flex items-center gap-1"
                  href="https://www.marketboro.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  마켓보로 웹사이트
                  <span>
                    ↗
                  </span>
                </a>
              </li>
              <li>
                <a
                  class="text-blue-600 hover:underline inline-flex items-center gap-1"
                  href="https://www.foodspring.co.kr"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  식봄
                  <span>
                    ↗
                  </span>
                </a>
              </li>
              <li>
                <a
                  class="text-blue-600 hover:underline inline-flex items-center gap-1"
                  href="https://pro.marketbom.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  마켓봄 프로
                  <span>
                    ↗
                  </span>
                </a>
              </li>
            </ul>
          </section>
        </li>
        <li
          class="p-6 rounded-xl border shadow-sm bg-background text-foreground flex flex-col gap-4"
        >
          <div
            class="flex items-start justify-start gap-x-4"
          >
            <img
              alt="(주) 코렌스 로고"
              class="rounded-full size-12 border"
              src="/src/assets/logos/korens_logo.webp"
            />
            <div>
              <h2
                class="text-xl font-bold"
              >
                (주) 코렌스
              </h2>
              <p
                class="text-sm text-gray-500 dark:text-gray-200 mt-1"
              >
                프론트엔드 개발자
                 | 
                2019.07 ~ 2020.10
              </p>
            </div>
          </div>
          <section
            class="text-gray-700 dark:text-gray-300 leading-relaxed text-sm whitespace-pre-line"
          >
            (주) 코렌스는 자동차 부품 제조사로서, 소프트웨어 사업부를 신설하여 새로운 도전을 시작하였습니다. 
          이곳에서 프론트엔드 개발자로서의 커리어를 시작하였으며, 
          Vue.js를 활용한 프론트엔드 개발뿐만 아니라 백엔드 개발과 AWS Amplify를 통한 클라우드 서비스 구축 등 다양한 역할을 수행하였습니다. 
          특히, 초기 개발 환경과 프로세스 구축에 참여하여, 개발 역량뿐만 아니라 문제 해결 능력과 주도적인 업무 수행 능력을 함양하였습니다.
          </section>
          <section>
            <h6
              class="mt-2 font-semibold text-sm text-gray-700 dark:text-gray-300"
            >
              Projects
            </h6>
            <ul
              class="list-disc ml-5 text-sm mt-1"
            />
          </section>
          <section>
            <h6
              class="mt-2 font-semibold text-sm text-gray-700 dark:text-gray-300"
            >
              Links
            </h6>
            <ul
              class="list-disc ml-5 text-sm mt-1 space-y-1"
            >
              <li>
                <a
                  class="text-blue-600 hover:underline inline-flex items-center gap-1"
                  href="http://www.korens.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  코렌스 웹사이트
                  <span>
                    ↗
                  </span>
                </a>
              </li>
            </ul>
          </section>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="flex flex-col gap-8 pt-8"
    >
      <li
        class="p-6 rounded-xl border shadow-sm bg-background text-foreground flex flex-col gap-4"
      >
        <div
          class="flex items-start justify-start gap-x-4"
        >
          <img
            alt="(주) 마켓보로 로고"
            class="rounded-full size-12 border"
            src="/src/assets/logos/marketboro_logo.webp"
          />
          <div>
            <h2
              class="text-xl font-bold"
            >
              (주) 마켓보로
            </h2>
            <p
              class="text-sm text-gray-500 dark:text-gray-200 mt-1"
            >
              프론트엔드 개발자
               | 
              2020.10 ~ 
            </p>
          </div>
        </div>
        <section
          class="text-gray-700 dark:text-gray-300 leading-relaxed text-sm whitespace-pre-line"
        >
          (주) 마켓보로는 B2B 식자재 유통 시장에서 중간 과정을 최소화하여 비용 절감과 효율 증대를 실현하는 엔터프라이즈 푸드테크 플랫폼을 제공합니다. 
          또한 사업자 대상의 오픈 마켓인 식봄을 운영하고 있습니다. 
          저는 프론트엔드 개발자로서 React.js와 Next.js를 활용하여 웹과 모바일 애플리케이션의 사용자 인터페이스를 개발하고 있습니다. 
          특히, 기존 PHP로 개발된 식봄을 Next.js로 전환하는 프로젝트를 주도하며, 성능 최적화와 사용자 경험 향상을 달성하였습니다. 
          이러한 경험을 통해 다양한 디바이스에서의 접근성을 보장하고, 최신 기술 스택을 적극 활용하는 역량을 갖추게 되었습니다.
        </section>
        <section>
          <h6
            class="mt-2 font-semibold text-sm text-gray-700 dark:text-gray-300"
          >
            Projects
          </h6>
          <ul
            class="list-disc ml-5 text-sm mt-1"
          />
        </section>
        <section>
          <h6
            class="mt-2 font-semibold text-sm text-gray-700 dark:text-gray-300"
          >
            Links
          </h6>
          <ul
            class="list-disc ml-5 text-sm mt-1 space-y-1"
          >
            <li>
              <a
                class="text-blue-600 hover:underline inline-flex items-center gap-1"
                href="https://www.marketboro.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                마켓보로 웹사이트
                <span>
                  ↗
                </span>
              </a>
            </li>
            <li>
              <a
                class="text-blue-600 hover:underline inline-flex items-center gap-1"
                href="https://www.foodspring.co.kr"
                rel="noopener noreferrer"
                target="_blank"
              >
                식봄
                <span>
                  ↗
                </span>
              </a>
            </li>
            <li>
              <a
                class="text-blue-600 hover:underline inline-flex items-center gap-1"
                href="https://pro.marketbom.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                마켓봄 프로
                <span>
                  ↗
                </span>
              </a>
            </li>
          </ul>
        </section>
      </li>
      <li
        class="p-6 rounded-xl border shadow-sm bg-background text-foreground flex flex-col gap-4"
      >
        <div
          class="flex items-start justify-start gap-x-4"
        >
          <img
            alt="(주) 코렌스 로고"
            class="rounded-full size-12 border"
            src="/src/assets/logos/korens_logo.webp"
          />
          <div>
            <h2
              class="text-xl font-bold"
            >
              (주) 코렌스
            </h2>
            <p
              class="text-sm text-gray-500 dark:text-gray-200 mt-1"
            >
              프론트엔드 개발자
               | 
              2019.07 ~ 2020.10
            </p>
          </div>
        </div>
        <section
          class="text-gray-700 dark:text-gray-300 leading-relaxed text-sm whitespace-pre-line"
        >
          (주) 코렌스는 자동차 부품 제조사로서, 소프트웨어 사업부를 신설하여 새로운 도전을 시작하였습니다. 
          이곳에서 프론트엔드 개발자로서의 커리어를 시작하였으며, 
          Vue.js를 활용한 프론트엔드 개발뿐만 아니라 백엔드 개발과 AWS Amplify를 통한 클라우드 서비스 구축 등 다양한 역할을 수행하였습니다. 
          특히, 초기 개발 환경과 프로세스 구축에 참여하여, 개발 역량뿐만 아니라 문제 해결 능력과 주도적인 업무 수행 능력을 함양하였습니다.
        </section>
        <section>
          <h6
            class="mt-2 font-semibold text-sm text-gray-700 dark:text-gray-300"
          >
            Projects
          </h6>
          <ul
            class="list-disc ml-5 text-sm mt-1"
          />
        </section>
        <section>
          <h6
            class="mt-2 font-semibold text-sm text-gray-700 dark:text-gray-300"
          >
            Links
          </h6>
          <ul
            class="list-disc ml-5 text-sm mt-1 space-y-1"
          >
            <li>
              <a
                class="text-blue-600 hover:underline inline-flex items-center gap-1"
                href="http://www.korens.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                코렌스 웹사이트
                <span>
                  ↗
                </span>
              </a>
            </li>
          </ul>
        </section>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
